src/site/SiteNavigation.js

```diff
@@ -0,0 +1,215 @@
+import {
+  AddClassToElms,
+  RemoveClassFromElms,
+  GetFirstElementOrDefault,
+  SetAttributeForElms
+} from "../utilities/dom.utilities";
+import FocusTrap from "focus-trap";
+
+const attributes = {
+  ariaExpanded: "aria-expanded",
+  tabIndex: "tabindex"
+};
+
+const icons = {
+  open: "fa-bars",
+  close: "fa-times"
+};
+
+const ids = {
+  siteNav: "bc_site-nav",
+  siteNavToggleButton: "bc_site-nav__toggle-button",
+  siteNavLinks: "bc_site-nav__links",
+  page: "bc_page"
+};
+
+const cssClasses = {
+  hidden: "hidden",
+  hiddenInit: "hidden-init",
+  isActive: "is-active",
+  isDisabled: "is-disabled",
+  siteNavToggleButtonText: "bc_site-nav__toggle-button__text"
+};
+
+const invisibleTabIndex = "-1"; /** Allows us to hide an item from tabbing */
+let siteNavFocusTrap;
+
+/**
+ * * Hide/Show the site navigation based on the given shouldShow param
+ * @param {boolean} shouldShow if set to true the site navigation will be shown
+ */
+const toggleSiteNav = shouldShow => {
+  const classListAction = shouldShow ? "add" : "remove";
+  const siteNavToggleButtonElm = document.getElementById(
+    ids.siteNavToggleButton
+  );
+  const buttonIconElm = GetFirstElementOrDefault(siteNavToggleButtonElm, "i");
+  const buttonIconTextElm = GetFirstElementOrDefault(
+    siteNavToggleButtonElm,
+    `.${cssClasses.siteNavToggleButtonText}`
+  );
+  const pageElm = document.getElementById(ids.page);
+
+  // Update toggle button text
+  buttonIconTextElm.textContent = shouldShow ? "Close" : "Menu";
+
+  // Toggle the toggle button icon
+  buttonIconElm.classList.remove(shouldShow ? icons.open : icons.close);
+  buttonIconElm.classList.add(shouldShow ? icons.close : icons.open);
+
+  // Set / Remove aria-expanded attributes
+  SetAttributeForElms(
+    [siteNavToggleButtonElm, pageElm],
+    attributes.ariaExpanded,
+    shouldShow
+  );
+
+  // Set / Remove tab-index inside Site Nav
+  updateSiteNavTabIndex(shouldShow);
+
+  // Activate or deactivate proper focus state
+  siteNavFocusTrap[shouldShow ? "activate" : "deactivate"]();
+
+  // Add / Remove esc listener
+  document[shouldShow ? "addEventListener" : "removeEventListener"](
+    "keydown",
+    handleDocumentKeyDown,
+    true
+  );
+
+  // Hide / Show the site nav
+  document
+    .getElementById(ids.siteNav)
+    .classList[classListAction](cssClasses.isActive);
+
+  // Enable / Disabled page and body
+  const disabledElms = [pageElm, document.body];
+  if (shouldShow) {
+    AddClassToElms(disabledElms, cssClasses.isDisabled);
+  } else {
+    RemoveClassFromElms(disabledElms, cssClasses.isDisabled);
+  }
+};
```

Comment:
In this case could you instead use ToggleClassForElms(disabledElms, cssClasses.isDisabled)?

Reply:
The way this is designed to toggle is based on the given flag passed into the fucntion (so may be the function name is bad), but for that reason we can't toggle.

Reply:
Okay, I see that.